lighthouse-viewer/gulpfile.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const del = require('del');
+const gulp = require('gulp');
+const gulpLoadPlugins = require('gulp-load-plugins');
+const runSequence = require('run-sequence');
+const browserify = require('browserify');
+const closure = require('google-closure-compiler-js').gulp();
+const ghPages = require('gulp-gh-pages');
+
+const $ = gulpLoadPlugins();
+
+function license() {
+  return $.license('Apache', {
+    organization: 'Copyright 2016 Google Inc. All rights reserved.',
+    tiny: true
+  });
+}
+
+function applyBrowserifyTransforms(bundle) {
+  // Fix an issue with imported speedline code that doesn't brfs well.
+  return bundle.transform('../lighthouse-extension/fs-transform', {
+    global: true
+  }).transform('brfs', {global: true}); // Transform the fs.readFile etc. Do so in all the modules.
+}
+
+gulp.task('lint', () => {
+  return gulp.src([
+    'app/src/**/*.js',
+    'gulpfile.js'
+  ])
+  .pipe($.eslint())
+  .pipe($.eslint.format());
+});
+
+gulp.task('images', () => {
+  return gulp.src('app/images/**/*')
+  .pipe(gulp.dest('dist/images'));
+});
+
+gulp.task('css', () => {
+  return gulp.src([
+    'app/styles/**/*.css',
+    '../lighthouse-core/report/styles/report.css'
+  ])
+  .pipe(gulp.dest('dist/styles'));
+});
+
+gulp.task('html', () => {
+  return gulp.src('app/*.html').pipe(gulp.dest('dist'));
+});
+
+gulp.task('browserify', () => {
+  return gulp.src([
+    'app/src/main.js'
+  ], {read: false})
+    .pipe($.tap(file => {
+      let bundle = browserify(file.path);
+      bundle = applyBrowserifyTransforms(bundle);
+
+      // Inject transformed browserified content back into our gulp pipeline.
+      file.contents = bundle.bundle();
+    }))
+    .pipe(gulp.dest('dist/src'));
+});
+
+gulp.task('compile', ['browserify'], () => {
+  return gulp.src([
+    'dist/src/main.js'
+  ])
+    .pipe($.sourcemaps.init())
+    .pipe(closure({
+      compilationLevel: 'SIMPLE',
+      // warningLevel: 'VERBOSE',
+      // outputWrapper: '(function(){\n%output%\n}).call(this)',
+      // languageOut: 'ECMASCRIPT5',
+      // processCommonJsModules: true,
+      jsOutputFile: 'main.js',
+      createSourceMap: true
+    }))
+    .pipe($.uglify()) // Use uglify to strip out duplicated license headers.
```

Comment:
Yea :\

Closure unfortunately leaves a bunch of licenses dupes at the top of the file:

<img width="491" alt="screen shot 2016-12-05 at 5 28 16 pm" src="https://cloud.githubusercontent.com/assets/238208/20909393/504a64ca-bb10-11e6-9fa9-e4eed4b69754.png">

After some reading, I think it may be because we use `@license` in our headers and Closure explicitly keeps them them around. I introduced uglify to remove all comments and add back the license.

I can switch to babel, but I'd prefer to do that in another PR. At the moment, closure produces the smallest filesize and transpiles for us. I'm also interested in using advanced compilation later for tree shaking out some of the browserified cruft. 

Build time shouldn't be an issue. For dev, `gulp browserify` (or `gulp watch`) give you fast builds. For prod/deploy, `gulp compile` takes 34s for me, but it's a one time thing.